src/main/java/org/elasticsearch/index/query/BaseQueryBuilder.java

```diff
@@ -79,4 +79,10 @@ public Query toQuery(QueryParseContext parseContext) throws QueryParsingExceptio
     protected abstract String parserName();
 
     protected abstract void doXContent(XContentBuilder builder, Params params) throws IOException;
+
+    @Override
+    public void validate() {
+        // default impl does not validate, subclasses should override.
+        // TODO Remove once all queries support validation
+    };
```

Comment:
sorry I keep getting confused between parser and builder, we do need it. can you remove the `;` at the end?